File: stock-app/stock-app/PortfolioItem.swift
//
//  PortfolioItem.swift
//  stock-app
//
//  Created by Varun Mehta on 08/04/24.
//

// PortfolioItem.swift

import Foundation

struct PortfolioItem: Codable {
   
    let symbol: String
    let name: String
    let quantity: Int
    let totalCost: Double
    let averageValuePerShare: Double
    let change: Double
    let currentPrice: Double
    let marketValue: Double
}

struct PortfolioRepositories:Codable{
    let portfolios:[PortfolioItem]
}

File: backend/model/PortfolioItem.js
const mongoose = require('mongoose');

const portfolioSchema = new mongoose.Schema({
  symbol: {
    type: String,
    required: true
  },

    name: {
    type: String,
    required: true
  },


  quantity: {
    type: Number,
    required: true
  },
  totalCost: {
    type: Number,
    required: true
  },
  averageValuePerShare: {
    type: Number,
    required: true
  },
  change: {
    type: Number,
    required: true
  },
  
  currentPrice: {
    type: Number,
    required: true
  },
  marketValue: {
    type: Number,
    required: true
  }
});

const PortfolioItem = mongoose.model('Portfolio', portfolioSchema);

module.exports = PortfolioItem;
